lib/data_update_scripts/20210329172336_remove_pro_roles.rb

```diff
@@ -0,0 +1,12 @@
+module DataUpdateScripts
+  class RemoveProRoles
+    def run
+      pro_role = Role.find_by(name: "pro")
+
+      return unless pro_role
+
+      pro_role.users.find_each { |u| u.remove_role(:pro) }
+      pro_role.destroy
+    end
+  end
+end
```

Comment:
Do we have any concerns about running this on Forem instances other than DEV? Should we add a guard clause if so?

Reply:
That is, I don't know how roles are defined and if another Forem instance defined a pro role on purpose, we may not want to stomp on that.

If the roles are defined in code, then we probably have nothing to worry about. I'm just unsure where we get the names from. 😄

Reply:
@vaidehijoshi Oooh right, sorry, I missed that here. Super appreciate the insight, though! As long as Forem admins can't define custom roles which might happen to be `pro`, this looks really well done! 💯

Reply:
Sure thing, @jgaskins! Glad you're thinking about potential failure modes, that's always a good thing 😉 

If someone wanted to add a new role (or remove a role, like what we're doing 😃), they'd have to commit a code change to the `Role` model, so I don't think that specific edge case can easily happen to any Forem admins (unless someone forks the repo of course).

Reply:
@vaidehijoshi We have a foreign key constraint that cascades:

```ruby
add_foreign_key "users_roles", "roles", on_delete: :cascade
```

So removing the role would also delete it from the join table. I tried locally and it worked fine for me:

```ruby
pro_role = Role.find_by(name: "pro")
return unless pro_role

pro_role.destroy
```

That said, I'm not sure what other magic the `rolify` gem performs, so maybe it's better to stick to `remove_role` since we're only dealing with a little over a hundred users here.

Reply:
This is the magic

https://github.com/RolifyCommunity/rolify/blob/61d90f7ccae569fe18e54a7776e5d34270380898/lib/rolify/adapters/active_record/role_adapter.rb#L58-L70

I haven't dug in further, so I think we should leave the business logic code as well

Reply:
Ok! So it sounds like (based on @rhymes's last comment) I can leave the script the way that I have it